src/node.cc

```diff
@@ -2498,36 +2502,96 @@ struct node_module* get_linked_module(const char* name) {
   return mp;
 }
 
-// DLOpen is process.dlopen(module, filename).
+struct DLib {
```

Comment:
Should this be a `struct`? If we keep it as a `class` and accept the values of `filename_` and `errmsg_` in the constructor, then we don't have to expose all the members as `public` right?

Reply:
Already discussed. See above.

Reply:
I couldn't find the actual discussion corresponding to this. I know it is exhausting to search through it, but if you could find it, please point me to it. I'll try again when I get back home.

Reply:
On https://github.com/nodejs/node/pull/12794#discussion_r124620884 it was suggested to do a class. So I did it and commented 
 https://github.com/nodejs/node/pull/12794#issuecomment-312984056. Later, https://github.com/nodejs/node/pull/12794#discussion_r128185744, where it was suggested to do a struct instead.

Reply:
Sorry, I didn't check properly then.